### ISSUE TITLE
WebSocket connection failure and kernel disconnection

### ISSUE THREAD

**Issue by vivuvivek** [jupyter/notebook#3382]
We are using Jupyter notebook installed on HDInsight spark cluster. While executing long running notebook, using FQDN URL, we are facing WebSocket connection failure and kernel disconnection

If we use internal URL of the head node to connect, whole process works without any issue.

Screen shot of the error log from Firefox debug window:
![git_issue](https://user-images.githubusercontent.com/4204037/36772958-49e25024-1c4f-11e8-98a8-779cc69d2303.png)

Is there a proxy setting(related to timeout) that we need to configure, if we are accessing the notebook behind a proxy/gateway?

Total running time of the sample notebook that we are testing is ~20 mins

**Comment by takluyver** [comment]
Yes, some proxies cut the websocket connection after a certain amount of time. I don't know exactly what setting is involved, though. It presumably depends on your proxy.

By default, the notebook sends a 'ping' on the websocket every 30 seconds. You can configure this to be lower with `NotebookApp.tornado_settings['ws_ping_interval']` (which should be set to a number in milliseconds).

**Comment by rwmajor2** [comment]
@takluyver Can we see this 'ping' in a web browser dev tools or something other means to validate?

**Comment by bollwyvl** [comment]
The browser probably won't tell you... a tool like `wireshark` or `mitmproxy` would, howver.
